--- a/dev/null	2019-05-14 23:06:29.539627306 -0500
+++ b/package/trx.mk	2019-05-28 21:26:15.576085497 -0500
@@ -0,0 +1,28 @@
+################################################################################
+#
+# trx
+#
+################################################################################
+
+TRX_VERSION =  0.3
+TRX_SOURCE = trx-$(TRX_VERSION).tar.gz
+TRX_SITE= http://www.pogo.org.uk/~mark/trx/releases
+TRX_INSTALL_STAGING = YES
+TRX_INSTALL_TARGET = YES
+TRX_AUTORECONF = YES 
+#TRX_CONF_OPTS = --disable-portaudio --disable-portmidi --no-recursion
+
+TRX_DEPENDENCIES = opus ortp
+
+
+define TRX_BUILD_CMDS
+     $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+define TRX_INSTALL_TARGET_CMDS
+	$(INSTALL) -D -m 0755 $(@D)/tx $(TARGET_DIR)/usr/bin
+	$(INSTALL) -D -m 0755 $(@D)/tx $(TARGET_DIR)/usr/bin
+endef
+
+
+$(eval $(generic-package))


--- a/dev/null	2019-05-14 23:06:29.539627306 -0500
+++ b/package/Config.in	2019-05-28 21:24:48.833463902 -0500
@@ -0,0 +1,8 @@
+config BR2_PACKAGE_TRX
+	bool "trx"
+	select BR2_PACKAGE_OPUS
+	select BR2_PACKAGE_OPUSFILE
+	
+	help
+		realtime audio stream lib
+

--- a/dev/null	2019-05-14 23:06:29.539627306 -0500
+++ b/package/0001-null.patch	2019-05-28 21:29:56.948495708 -0500
@@ -0,0 +1,12 @@
+--- a/tx.c	2019-05-28 21:27:29.898892270 -0500
++++ b/tx.c	2019-05-28 21:27:47.966595516 -0500
+@@ -240,7 +240,7 @@
+ 
+ 	ortp_init();
+ 	ortp_scheduler_init();
+-	ortp_set_log_level_mask(ORTP_WARNING|ORTP_ERROR);
++	ortp_set_log_level_mask(NULL, ORTP_WARNING|ORTP_ERROR);
+ 	session = create_rtp_send(addr, port);
+ 	assert(session != NULL);
+ 
+
