--- a/dev/null	2019-05-14 23:06:29.539627306 -0500
+++ b/package/orca.mk	2019-05-20 12:58:14.151411265 -0500
@@ -0,0 +1,24 @@
+################################################################################
+#
+# orca
+#
+################################################################################
+
+ORCA_VERSION =  df0e1276ec2ef795f8a072a6f6768cc7ef73647c
+ORCA_SITE = $(call github,hundredrabbits,Orca-c,$(ORCA_VERSION))
+ORCA_INSTALL_TARGET = YES
+
+ORCA_DEPENDENCIES = ncurses
+
+
+define ORCA_BUILD_CMDS
+$(MAKE) CC="$(TARGET_CC)" LD=$(TARGET_LD) -C $(@D) release
+endef
+
+define ORCA_INSTALL_TARGET_CMDS
+$(INSTALL) -D -m 0755 $(@D)/build/release/orca $(TARGET_DIR)/usr/bin
+endef
+
+#$(INSTALL) -D -m 0644 $(@D)/examples/* $(TARGET_DIR)/usr/share/orca
+
+$(eval $(generic-package))
raul@vurtek:~/feb25/HackadaySynth/buildroot-2017.11-rc1/package/orca$ diff -urN /dev/null Config.in 
--- a/dev/null	2019-05-14 23:06:29.539627306 -0500
+++ b/package/Config.in	2019-05-20 10:23:22.307106360 -0500
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_ORCA
+	bool "orca"
+	select BR2_PACKAGE_NCURSES
+	
+	help
+		ncurses atypical sequencer
+

