--- a/board/friendlyarm/nanopi-neo/boot.cmd	2019-12-22 10:36:29.416890693 -0600
+++ b/board/friendlyarm/nanopi-neo/boot.cmd	2019-12-22 10:37:09.155777996 -0600
@@ -1,6 +1,6 @@
 setenv fdt_high ffffffff
 
-setenv bootargs console=ttyS0,115200 earlyprintk root=/dev/mmcblk0p2 rootwait
+setenv bootargs console=ttyS0,115200 earlyprintk root=/dev/mmcblk0p2 rootwait isolcpus=2,3 sunxi_ve_mem_reserve=0 sunxi_g2d_mem_reserve=0 sunxi_no_mali_mem_reserve sunxi_fb_mem_reserve=16 consoleblank=0 nohz=off
 
 fatload mmc 0 $kernel_addr_r zImage
 fatload mmc 0 $fdt_addr_r sun8i-h3-nanopi-neo.dtb

