--- a/arch/arm/boot/dts/sun8i-h3-nanopi-neo.dts	2018-07-08 22:12:40.469431970 -0500
+++ b/arch/arm/boot/dts/sun8i-h3-nanopi-neo.dts	2018-07-08 22:13:24.964853556 -0500
@@ -46,6 +46,13 @@
 	model = "FriendlyARM NanoPi NEO";
 	compatible = "friendlyarm,nanopi-neo", "allwinner,sun8i-h3";
 };
+
+&uart1 {
+        pinctrl-names = "default";
+        pinctrl-0 = <&uart1_pins>;
+        status = "okay";
+};
+
 &usb_otg {
 	dr_mode = "otg";
 	status = "okay";
@@ -103,4 +114,10 @@
 	status = "okay";
 };
 
-
+&codec {
+	allwinner,audio-routing =
+		"Line Out", "LINEOUT",
+		"MIC1", "Mic",
+		"Mic",  "MBIAS";
+	status = "okay";
+};


+&i2c0 {
+	status = "okay";
+};
+
